include/clasp/core/bignum.h

```diff
@@ -69,35 +82,55 @@ class Bignum_O : public Integer_O {
     return b;
   };
   static Bignum_sp create(const mpz_class&);
-  static Bignum_sp create(gc::Fixnum fix) { return create_from_limbs((fix < 0) ? -1 : 1, std::abs(fix), true); }
-#if !defined(CLASP_UNSIGNED_LONG_LONG_IS_UINT64)
-  static Bignum_sp create(unsigned long long ull) {
-    ASSERT(sizeof(unsigned long long) <= sizeof(mp_limb_t));
-    return create_from_limbs(1, ull, true);
+
+  template <unsigned_limb T> static Bignum_sp create(T v) {
+    Bignum_sp b = create_from_limbs(1);
+    b->_limbs[0] = v;
+    return b;
   }
-#endif
-#if !defined(CLASP_LONG_LONG_IS_INT64)
-  static Bignum_sp create(long long ll) {
-    ASSERT(sizeof(long long) <= sizeof(mp_limb_t));
-    return create_from_limbs((ll < 0) ? -1 : 1, std::abs(ll), true);
+
+  template <signed_limb T> static Bignum_sp create(T v) {
+    Bignum_sp b = create_from_limbs((v < 0) ? -1 : 1);
+    b->_limbs[0] = std::abs(v);
+    return b;
   }
-#endif
-#if !defined(CLASP_FIXNUM_IS_INT64)
-  static Bignum_sp create(int64_t v) { return create_from_limbs((v < 0) ? -1 : 1, std::abs(v), true); }
-#endif
-  static Bignum_sp create(uint64_t v) { return create_from_limbs(1, v, true); }
-  static Bignum_sp create(__uint128_t v) {
-    Bignum_sp b = create_from_limbs(2);
-    b->_limbs[0] = static_cast<mp_limb_t>(v);
-    b->_limbs[1] = static_cast<mp_limb_t>(v >> 64);
+
+  template <unsigned_limbs T> static Bignum_sp create(T v) {
+    constexpr size_t limb_width = 8 * sizeof(mp_limb_t);
```

Comment:
obviously it's not going to matter conformancewise, but i think we should use CHAR_BIT instead of 8 just to get the point across.

Reply:
Good point